src/db.rs

```diff
@@ -1,297 +1,533 @@
+    let res: anyhow::Result<BotConfig> = sqlx::query_as::<_, DBBotConfig>(
+        "SELECT genesis_qr, invite_qr, tester_group, reviewee_group, genesis_group, serial FROM config",
+    )
+    .fetch_one(c)
+    .await
+    .map(|db_bot_config| db_bot_config.try_into())?;
+    res
+}
 
-    pub async fn create_submit(&self, chat: &SubmitChat) -> anyhow::Result<()> {
-        let res: Option<SubmitChat> = self
-            .db
-            .create(("chat", chat.creator_chat.to_u32().to_string()))
-            .content(chat)
-            .await?;
-        Ok(())
-    }
+pub async fn create_submit_chat(c: &mut SqliteConnection, chat: &SubmitChat) -> sqlx::Result<()> {
+    sqlx::query("INSERT INTO chats (submit_chat_id, submit_helper, app_info) VALUES (?, ?, ?)")
+        .bind(chat.submit_chat.to_u32())
+        .bind(chat.submit_helper.to_u32())
+        .bind(chat.app_info)
+        .execute(c)
+        .await?;
+    Ok(())
+}
 
-    pub async fn upgrade_to_review_chat(&self, chat: &ReviewChat) -> surrealdb::Result<()> {
-        let res: Option<ReviewChat> = self
-            .db
-            .update(("chat", chat.creator_chat.to_u32().to_string()))
-            .content(chat)
-            .await?;
-        Ok(())
-    }
+/// Upgrade a submit chat with chat_id `id` to a review chat
+pub async fn upgrade_to_review_chat(
+    c: &mut SqliteConnection,
+    chat: &ReviewChat,
+) -> anyhow::Result<()> {
+    sqlx::query(
+        "UPDATE chats SET review_helper = ?, submit_helper = ?, review_chat_id = ?, submit_chat_id = ?, publisher = ?, app_info = ? WHERE submit_chat_id = ?"
+    ).bind(chat.review_helper.to_u32())
+    .bind(chat.submit_helper.to_u32())
+    .bind(chat.submit_chat.to_u32())
+    .bind(chat.publisher.to_u32())
+    .bind(chat.submit_chat.to_u32())
+    .bind(chat.review_chat.to_u32()).execute(c).await?;
+    Ok(())
+}
 
-    pub async fn set_chat_type(
-        &self,
-        chat_id: ChatId,
-        chat_type: ChatType,
-    ) -> surrealdb::Result<()> {
-        let _t: Option<DBChatType> = self
-            .db
-            .create(("chattype", chat_id.to_u32().to_string()))
-            .content(DBChatType { chat_type })
-            .await?;
-        Ok(())
-    }
+pub async fn get_review_chat(
+    c: &mut SqliteConnection,
+    chat_id: ChatId,
+) -> anyhow::Result<ReviewChat> {
+    sqlx::query("SELECT review_helper, submit_helper, review_chat_id, submit_chat_id, publisher, app_info FROM chats WHERE (review_chat_id = ?)")
+        .bind(chat_id.to_u32())
+        .fetch_one(c)
+        .await
+        .map(|row| {
+            Ok(ReviewChat {
+                review_helper: MsgId::new(row.try_get("review_helper")?),
+                submit_helper: MsgId::new(row.try_get("submit_helper")?),
+                review_chat: ChatId::new(row.try_get("review_chat_id")?),
+                submit_chat: ChatId::new(row.try_get("submit_chat_id")?),
+                publisher: ContactId::new(row.try_get("publisher")?),
```

Comment:
Is it not possible to implement some traits for `MsgId`,`ContactId`and `ChatId` so that sqlx can do this automatically for you?